packages/connect/src/webrtc/connection.ts

```diff
@@ -405,6 +447,8 @@ class WebRTCConnection implements MultiaddrConnection {
    * available, yield messages from WebRTC instance
    */
   private async *createSource(): StreamSource {
```

Comment:
Also might be broken up to two or 3 smaller functions.

Reply:
postponed for future refactor